### PR TITLE
feat: automate all calendar sync popups for iOS

### PR DESCRIPTION
**Pull request by bilalawan321**
[LEARNER-8998](https://2u-internal.atlassian.net/browse/LEARNER-8998)

**Description**

Following test cases will be implemented:
Verify that Calendar sync toggle is working properly
Verify that on switching on the toggle a pop up appears (edx whould like to access your calendar) with "Don't Allow" & "OK" button
Verify that tapping on "Don't Allow" will close the pop-up and the toggle will be switched Off
Verify that Settings pop-up appears after tapping "Don't Allow" button and switching the toggle back On
Verify that user is able to get permission for calendar after tapping open settings button
Verify that tapping on Cancel button will close the pop-up
Verify that tapping on OK button from access your calendar pop-up an other pop-up appears
Verify that toggle is switched on after tapping Done button
Verify that a pop-up appears with (Cancel & Remove) button after switching off the toggle
Verify that Sync Calendar toggle remains On after tapping cancel button from the pop-up

### CODE REVIEW

File: tests/ios/tests/test_ios_course_dates.py [hunk 1]
```diff
@@ -71,6 +71,147 @@ def test_validate_ui_elements_smoke(self, set_capabilities, setup_logging):
...
+            set_capabilities,
+            ios_elements.dates_calendar_sync_toggle)
+        assert sync_calendar_toggle.text == strings.TOGGLE_ON_IOS
+
+    def test_calendar_remove_events_alert_smoke(self, set_capabilities, setup_logging):
+        """
+        Scenarios:
+            Verify that a pop-up appears with (Cancel & Remove) button after switching off the toggle
+            "Your course calendar has been removed" toast appears after switching off the toggle and
+                tapping on remove button
```
Comment (moeez96) [inline tests/ios/tests/test_ios_course_dates.py:176]: Can we retain similar formatting of docstring project wide?
There seems to be a 4 spaces here in the beginning of the second whereas no spaces present in other docstrings added in this commit.

Comment (bilalawan321) [reply tests/ios/tests/test_ios_course_dates.py:176]: Hi, @moeez96.  yes, this formatting is the same in the whole project. 
you can see these 4 spaces from lines (27-36) and line 79 of this page.

File: tests/ios/tests/test_ios_course_dates.py [hunk 2]
```diff
@@ -71,6 +71,147 @@ def test_validate_ui_elements_smoke(self, set_capabilities, setup_logging):
 
         assert all_text[1].text == strings.DATES_COURSE_STARTS_TITLE
 
+    def test_calendar_sync_toggle_smoke(self, set_capabilities, setup_logging):
+        """
+        Scenarios:
+        Verify that Calendar sync toggle is working properly
+        Verify that on switching on the toggle a pop up appears
+            (edx whould like to access your calendar) with "Don't Allow" & "OK" button
```
Comment (bilalawan321) [inline tests/ios/tests/test_ios_course_dates.py:79]: @moeez96 here there are also 4 spaces.

File: tests/ios/tests/test_ios_course_dates.py (hunk 1 above)
Comment (bilalawan321) [reply tests/ios/tests/test_ios_course_dates.py:176]: @moeez96 I understand your point now. I have fixed your concerns.

**Pull request merged**